community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseFactoryState.java

```diff
@@ -75,6 +81,11 @@ public void addKernelExtensions( Iterable<KernelExtensionFactory<?>> newKernelEx
         }
     }
 
+    public void addURLAccessRule( String protocol, URLAccessRule rule )
```

Comment:
Since this is public API, we need URLAccessRule to live in the API as well (eg. under `org.neo4j.graphdb.*`) if we take this approach.

However - the GraphDBAPI already exposes the ability to configure this via regular config, that'd also be in line with the current approach of not exposing config options directly on the factory, but rather use `GraphDatabaseSettings` as the go-to place for programmatic configuration. 

Happy to revisit that discussion, but since there's contention about changing the design there, perhaps we could do it in two steps (eg. first this PR without config in the factory API, then discuss a potential API approach change).